var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -348,6 +348,15 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
         working_dir="third_party/gloo",
     )
 
+    # PyTorch does not build on Linux >=6.0.3 (fixed in master)
+    # See: https://github.com/facebookincubator/gloo/issues/345
+    patch(
+        "https://github.com/facebookincubator/gloo/commit/10909297fedab0a680799211a299203e53515032.patch?full_index=1",
+        sha256="8e6e9a44e0533ba4303a95a651b1934e5d73632cab08cc7d5a9435e1e64aa424",
+        when="@:2.0.0",
```

Comment:
```suggestion
        when="@:1",
```
Version ranges are inclusive, PyTorch 2 hasn't been released yet but the release candidate includes this patch.

Confirmed that 1.0.0 supports this patch.

We also have a gloo package you could patch if you want. Not required though.